_modules/ext_module_qvm.py

```diff
@@ -941,8 +941,197 @@ def prefs(vmname, *varargs, **kwargs):
+            current_devices.append(
+                {'device_type': device_type, 'backend': device.backend_domain.name, 'dev_id': device.ident,
+                 'options': device.options})
+
+    # Return all current devices if a 'list' only was selected
+    if args.list is not None or not (args.attach or args.detach):
+        message = []
+        for device in current_devices:
+            msg_options = '(' + ', '.join('{}={}'.format(key, value) for key, value in device['options'].items()) + ')'
+            message.append(
+                '    ' + device['device_type'] + ':' + device['backend'] + ':' + device['dev_id'] + ' ' + msg_options)
+
+        return {'result': True, 'comment': '[ATTACHED]:\n' + '\n'.join(message)}
+
+    # Helping function for parsing devices in args.attach and args.detach
+    def parse_device(raw_dev):
+        """
+        Example:
+            args.attach = [{u'bridge:sys-bridge:bridge0': [{u'ip': u'192.168.0.1'}, {u'netmask': u'255.255.255.0'}]}]
+        """
+        inline_dev = list(raw_dev.keys())[0]
+        device = {'device_type': inline_dev.split(':')[0], 'backend': inline_dev.split(':')[1],
+                  'dev_id': inline_dev.split(':')[2], 'options': {}}
+        for opt in raw_dev[inline_dev]:
+            device['options'].update(opt)
+
+        return device
+
+    # post-process argparse 'attach' and 'detach' values converted from string
+    if args.attach:
+        args.attach = [parse_device(raw_dev) for raw_dev in json.loads(args.attach[0])]
+
+    if args.detach:
+        args.detach = [parse_device(raw_dev) for raw_dev in json.loads(args.detach[0])]
+
+    for device in args.attach:
+        device_type = device['device_type']
+
+        if device['options'].get('pci_strictreset', False):
+            device['options']['no-strict-reset'] = not device['options']['pci_strictreset']
+            del device['options']['pci_strictreset']
+
+        message_old = None
+        for a in args.vm.devices[device_type].assignments():
+            if a.ident == device['dev_id']:
+                current_assignment = a
+
+                if current_assignment.options.get('no-strict-reset', False) != device['options'].get('no-strict-reset',
+                                                                                                     False):
+                    # detach and attach again to adjust options
+                    # args.vm.devices[device_type].update_persistent(a, False)
+                    args.vm.devices[device_type].detach(current_assignment)
+                    msg_options = '(' + ', '.join(
+                        '{}={}'.format(key, value) for key, value in current_assignment.options.items()) + ')'
+                    message_old = '[ATTACHED] ' + msg_options
+                break
+
+        try:
+            if device_type == 'pci' and ('no-strict-reset' not in device['options']):
+                device['options']['no-strict-reset'] = True
```

Comment:
Why? Adding 'no-strict-reset' by default for every PCI device definitely is _not_ a good idea.

Reply:
Yes, it's a misunderstood of https://github.com/QubesOS/qubes-mgmt-salt-dom0-qvm/blob/e3c2c967290bf36c50ddb694edd69b257ca258cf/_modules/ext_module_qvm.py#L913
Up to the negation of the condition, I was thinking it was the default case.